src/hooks/useSearchHighlightAndScroll.ts

```diff
@@ -46,19 +48,28 @@ function useSearchHighlightAndScroll({searchResults, transactions, previousTrans
             .map((actions) => Object.keys(actions ?? {}))
             .flat();
 
-        if (searchTriggeredRef.current) {
+        // Only proceed if we have previous data to compare against
+        // This prevents triggering on initial data load
+        if (previousTransactionsIDs.length === 0 && previousReportActionsIDs.length === 0) {
             return;
         }
+
         const hasTransactionsIDsChange = !isEqual(transactionsIDs, previousTransactionsIDs);
         const hasReportActionsIDsChange = !isEqual(reportActionsIDs, previousReportActionsIDs);
 
-        // NOTE: This if statement should NOT assume report actions can only change
-        // in one type, i.e.: isChat && hasReportActionsIDsChange
-        // because they can also change in other types such as CONST.SEARCH.DATA_TYPES.EXPENSE.
-        // Assuming they can only change in one type leads to issues such as
-        // https://github.com/Expensify/App/issues/57605
         // Check if there is a change in the transactions or report actions list
         if ((!isChat && hasTransactionsIDsChange) || hasReportActionsIDsChange) {
+            // Check if the detected changes are actually missing from current search results
+            const currentSearchResultIDs = isChat ? extractReportActionIDsFromSearchResults(searchResults?.data ?? {}) : extractTransactionIDsFromSearchResults(searchResults?.data ?? {});
+
+            const newIDs = isChat ? reportActionsIDs : transactionsIDs;
+            const genuinelyNewIDs = newIDs.filter((id) => !currentSearchResultIDs.includes(id));
```

Comment:
I think we are extracting these reports at [another](https://github.com/Expensify/App/blob/51508d4a088c7c4c836dd036c19c1a33fe8f327f/src/hooks/useSearchHighlightAndScroll.ts#L95) place too. We can make it DRYer by extracting this logic out of the `useEffect` s and use it in both the hooks.

```js
const existingSearchResultIDs = useMemo(() => {
    if (!searchResults?.data) {
        return [];
    }
    return isChat
        ? extractReportActionIDsFromSearchResults(searchResults.data)
        : extractTransactionIDsFromSearchResults(searchResults.data);
}, [searchResults?.data, isChat]);
```

```suggestion
            const newIDs = isChat ? reportActionsIDs : transactionsIDs;
            const hasAGenuinelyNewID = newIDs.some((id) => !existingSearchResultIDs.includes(id));
```